pkg/k8s/apis/cilium.io/v2alpha1/bgp_peer_types.go

```diff
@@ -0,0 +1,153 @@
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec is the specification of the desired behavior of the CiliumBGPPeerConfig.
+	Spec CiliumBGPPeerConfigSpec `json:"spec,omitempty"`
+}
+
+type CiliumBGPPeerConfigSpec struct {
+	// Transport defines the BGP transport parameters for the peer.
+	//
+	// +kubebuilder:validation:Optional
+	Transport *CiliumBGPTransport `json:"transport,omitempty"`
+
+	// Timers defines the BGP timers for the peer.
+	//
+	// +kubebuilder:validation:Optional
+	Timers *CiliumBGPTimers `json:"timers,omitempty"`
+
+	// AuthSecretRef is the name of the secret to use to fetch a TCP
+	// authentication password for this peer.
+	// +kubebuilder:validation:Optional
+	AuthSecretRef *string `json:"authSecretRef,omitempty"`
+
+	// GracefulRestart defines graceful restart parameters which are negotiated
+	// with this peer. If empty / nil, the graceful restart capability is disabled.
+	//
+	// +kubebuilder:validation:Optional
+	GracefulRestart *CiliumBGPNeighborGracefulRestart `json:"gracefulRestart,omitempty"`
+
+	// EBGPMultihopTTL controls the multi-hop feature for eBGP peers.
+	// Its value defines the Time To Live (TTL) value used in BGP
+	// packets sent to the peer.
+	// The value 1 implies that eBGP multi-hop feature is disabled
+	// (only a single hop is allowed).
+	//
+	// This field is ignored for iBGP peers.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=255
+	// +kubebuilder:default=1
+	EBGPMultihop *int32 `json:"ebgpMultihop,omitempty"`
+
+	// Families, if provided, defines a set of AFI/SAFIs the speaker will
+	// negotiate with it's peer.
+	//
+	// If this slice is not provided the default families of IPv6 and IPv4
+	// will be provided.
+	//
+	// +kubebuilder:validation:Optional
+	Families []CiliumBGPFamily `json:"families,omitempty"`
+}
+
+// CiliumBGPFamily represents a AFI/SAFI address family pair.
+type CiliumBGPFamily struct {
+	// Afi is the Address Family Identifier (AFI) of the family.
+	//
+	// +kubebuilder:validation:Enum=ipv4;ipv6;l2vpn;ls;opaque
+	// +kubebuilder:validation:Required
+	Afi string `json:"afi"`
+
+	// Safi is the Subsequent Address Family Identifier (SAFI) of the family.
+	//
+	// +kubebuilder:validation:Enum=unicast;multicast;mpls_label;encapsulation;vpls;evpn;ls;sr_policy;mup;mpls_vpn;mpls_vpn_multicast;route_target_constraints;flowspec_unicast;flowspec_vpn;key_value
+	// +kubebuilder:validation:Required
+	Safi string `json:"safi"`
+
+	// Advertisements selects group of BGP Advertisement(s) to advertise for this family.
+	//
+	// +kubebuilder:validation:Optional
+	Advertisements *slimv1.LabelSelector `json:"advertisements,omitempty"`
+}
+
+// CiliumBGPTransport defines the BGP transport parameters for the peer.
+type CiliumBGPTransport struct {
+	// LocalPort is the local port to be used for the BGP session.
+	//
```

Comment:
nit: Although implied by `+kubebuilder:default=179`, it's not a bad idea to include a godoc that states the default, e.g. "// When unspecified, defaults to TCP port 179.". Note that this comment applies to all other optional fields that use this kubebuilder marker and the godocs do not explicitly state a default value.